### PR TITLE
Adapt to pymatgen 2023.07.20 update.

### PR DESCRIPTION
**Pull request by qchempku2017**
## Summary

* Add support to pymatgen 2023.07.20
* Upgrade numpy to 1.25.1
* [issue] test_orbits/test_fixed_values now fails because licabr_orbit's symmetry operation is now 24 instead of 12. Need to figure this out!

## Checklist

- [x] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
      Run [pycodestyle](https://pycodestyle.readthedocs.io/en/latest/) and [flake8](http://flake8.pycqa.org/en/latest/)
      on your local machine.
- [x] Docstrings have been added in the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
      Run [pydocstyle](http://www.pydocstyle.org/en/2.1.1/index.html) on your code.
- [x] Tests have been added for any new functionality or bug fixes.
- [X] All existing tests pass.

### CODE REVIEW

**Comment by lbluque** [comment]
Perhaps the issue with the number of symmetry operations is related to this  some months ago materialsproject/pymatgen#2727

**Comment by lbluque** [comment]
> Perhaps the issue with the number of symmetry operations is related to this PR made some months ago [materialsproject/pymatgen#2727](https://github.com/materialsproject/pymatgen/pull/2727)

Submitting a bug fix, see materialsproject/pymatgen#3179

**Comment by qchempku2017** [comment]
> Perhaps the issue with the number of symmetry operations is related to this PR made some months ago [materialsproject/pymatgen#2727](https://github.com/materialsproject/pymatgen/pull/2727)

Wonderful! Thank you so much for taking care of this!

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 numpy==1.24.4
 monty==2023.5.8
-pymatgen==2023.06.23
+pymatgen==2023.07.20
```
Comment (lbluque) [inline requirements.txt:3]: can you remove this pin, we will have to wait to the next release so the symmetry errors are fixed

File: tests/test_cofe/test_orbit.py [hunk 2]
```diff
@@ -428,6 +428,7 @@ def licabr_orbit(single_structure):
 
 def test_fixed_values(licabr_orbit):
     assert licabr_orbit.multiplicity == 4
+    # TODO: figure out this failure.
```
Comment (lbluque) [inline tests/test_cofe/test_orbit.py:431]: remove this comment pls

**Review by lbluque — CHANGES_REQUESTED** [review]

**Comment by lbluque** [comment]
Thanks a lot @qchempku2017 !

**Pull request merged**